client/src/Try/API.purs

```diff
@@ -53,8 +55,8 @@ instance decodeJsonCompileError :: DecodeJson CompileError where
         map OtherError $ decodeJson contents
       "CompilerErrors" ->
         map CompilerErrors $ traverse decodeJson =<< decodeJson contents
-      _ ->
-        Left "Tag must be one of: OtherError, CompilerErrors"
+      j ->
+        Left $ AtKey "tag" $ UnexpectedValue $ encodeJson j
```

Comment:
I'm not sure how to do that. Can you clarify?

Reply:
Hmm. It’s not really the right one, though,  because it’s more for annotating structure. There’s not really a proper error type for this, other than maybe TypeMismatch. But I think it’s useful to show what the allowed constructors are in the error message.

Reply:
This is why I don't like `JsonDecodeError`. You can't provide error messages that have actionable "how to fix it" instructions.

Reply:
So, how should I address this issue?

Reply:
Yea, I don't love it either. Wish I hadn't added it to argonaut proper.

One option is this:

```purs
AtKey "tag" $ TypeMismatch "Expected tag to be one of 'OtherError, CompilerErrors'"
```

but I'm fine leaving it with what you've written here, too. Your call :)